src/models/contact.model.js

```diff
@@ -77,23 +77,25 @@ module.exports = class Contact extends BaseModel {
     // Map the Contact to the corresponding Dynamics schema Contact object
     const dataObject = {
       firstname: this.firstName,
-      lastname: this.lastName
+      lastname: this.lastName,
+      telephone1: this.telephone,
+      emailaddress1: this.email
     }
 
     try {
       let query
-      if (!this.contactid) {
-        // New contact
-        query = 'contacts'
-        return await dynamicsDal.createItem(dataObject, query)
-      } else {
+      if (this.id) {
         // Update contact
         query = `contacts(${this.contactid})`
```

Comment:
Should this now be `contacts(${this.id})` 
?

Reply:
Another good spot. Further highlights the current `update()` test isn't testing it properly.